Build error if calling a closure assigned to variable named 'new'. No issues with calling functions named 'new'

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: macos, macOS, 12.5.1, 21G83
Processor: 8 cpus, 64bit, little endian, Apple M1
CC version: Apple clang version 13.1.6 (clang-1316.0.21.2.5)

V full version: V 0.3.1 4588bb4.4bd49a0

Git version: git version 2.37.3
Git vroot status: 0.3.1-8-g4bd49a01-dirty
.git/config present: true
thirdparty/tcc status: thirdparty-macos-arm64 173c526e

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg simplenewerror.v`
```v

fn ret_int() fn (int) int{
	return fn( i int) int {
		return i
	}
}

// fn new(i int)int{
// 	return i
// }

fn main() {
	// Fails
	new := ret_int()
	println(new(42))

	// Works if anything but 'new'
	// news := ret_int()
	// println(news(42))
	// println('call function new ${new(42)}')  //uncomment new function

}
```
**What did you expect to see?**

42

**What did you see instead?**
```
==================
   ^~~~~~~~~~~~~~~
/tmp/v_501/simplenewerror.14945464780154816415.tmp.c:12016:18: error: implicit declaration of function '_v_new' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
        println(int_str(_v_new(42)));
                        ^
1 warning and 1 error generated.
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

```
More complex closure example
```v
struct Toy{
	id int
	name string 
}

fn make_toy( i int ) fn ( string ) Toy{
	mut idcount := i
	return fn [mut idcount] (sname string) Toy {
		return Toy{id: idcount++, name: sname }
	} 
}

fn (toy Toy) str() string{
	return 'ID ${toy.id:04}: "${toy.name}"'
}

fn main() {
	// OK closure 'new_toy := make_toy(101)' 
	new_toy := make_toy(101)
	mut arr  := 'Gun, Fire Truck, Police Car'.split(',').map( new_toy(it.trim_space()))

	// Fail 'new := make_toy(101)' //compiles but error when 'new(string)' included
	// new := make_toy(101)
	// mut arr  := 'Gun, Fire Truck, Police Car'.split(',').map( new(it.trim_space()))

	println(arr)
}
```

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -339,7 +339,7 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 					''
 				}
 
-				g.write('$ret_styp ($msvc_call_conv*${g.get_ternary_name(ident.name)}) (')
+				g.write('$ret_styp ($msvc_call_conv*${c_name(g.get_ternary_name(ident.name))}) (')
 				def_pos := g.definitions.len
 				g.fn_decl_params(func.func.params, unsafe { nil }, false)
 				g.definitions.go_back(g.definitions.len - def_pos)

--- a/vlib/v/gen/c/assign.v
+++ b/vlib/v/gen/c/assign.v
@@ -339,7 +339,8 @@ fn (mut g Gen) assign_stmt(node_ ast.AssignStmt) {
 					''
 				}
 
-				g.write('$ret_styp ($msvc_call_conv*${c_name(g.get_ternary_name(ident.name))}) (')
+				fn_name := c_name(g.get_ternary_name(ident.name))
+				g.write('$ret_styp ($msvc_call_conv*$fn_name) (')
 				def_pos := g.definitions.len
 				g.fn_decl_params(func.func.params, unsafe { nil }, false)
 				g.definitions.go_back(g.definitions.len - def_pos)

--- a/vlib/v/gen/c/utils.v
+++ b/vlib/v/gen/c/utils.v
@@ -65,7 +65,7 @@ fn (mut g Gen) unwrap(typ ast.Type) Type {
 // generate function variable definition, e.g. `void (*var_name) (int, string)`
 fn (mut g Gen) fn_var_signature(return_type ast.Type, arg_types []ast.Type, var_name string) string {
 	ret_styp := g.typ(return_type)
-	mut sig := '$ret_styp (*$var_name) ('
+	mut sig := '$ret_styp (*${c_name(var_name)}) ('
 	for j, arg_typ in arg_types {
 		arg_sym := g.table.sym(arg_typ)
 		if arg_sym.info is ast.FnType {

--- a/vlib/v/tests/fn_var_name_using_reserved_test.v
+++ b/vlib/v/tests/fn_var_name_using_reserved_test.v
@@ -1,0 +1,11 @@
+fn ret_int() fn (int) int {
+	return fn (i int) int {
+		return i
+	}
+}
+
+fn test_fn_var_name_using_reserved() {
+	new := ret_int()
+	println(new(42))
+	assert new(42) == 42
+}

